sky/serve/replica_managers.py

```diff
@@ -601,7 +646,23 @@ def _download_and_stream_logs(info: ReplicaInfo):
         self._down_process_pool[replica_id] = p
 
     def scale_down(self, replica_id: int) -> None:
-        self._terminate_replica(replica_id, sync_down_logs=False)
+
+        if replica_id in self._launch_process_pool:
+            p = self._launch_process_pool[replica_id]
+            if p.is_alive():
+                assert p.pid is not None
+                p.terminate()
+                p.join()
+            logger.info(f'Interrupted launch process for replica {replica_id} '
+                        'and deleted the cluster.')
+            self._terminate_replica(replica_id, sync_down_logs=False)
+            info = serve_state.get_replica_info_from_id(self._service_name,
+                                                        replica_id)
+            assert info is not None
+            info.status_property.sky_launch_status = ProcessStatus.INTERRUPTED
+            del self._launch_process_pool[replica_id]
```

Comment:
Should we move that logic into the `_terminate_replica` function?

Reply:
okay!